torch/onnx/symbolic_helper.py

```diff
@@ -437,6 +437,27 @@ def assert_training_mode(op_mode, op_name):
                       op_mode + " mode. The model will be exported in " +
                       training_mode + ", as specified by the export mode.")
 
+def _flatten_helper(g, input, start_dim, end_dim, dim):
+    # use Reshape for cases where the output shape is not 2D
+    if not input.isCompleteTensor():
+        return _unimplemented("flatten",
+                              "input size not accessible "
+                              "(consider using reshape op instead of flatten op to export to ONNX)")
+
+    input_size = g.op("Shape", input)
+    slice1 = _slice_helper(g, input_size, axes=[0], starts=[0], ends=[start_dim])
+    slice2 = _slice_helper(g, input_size, axes=[0], starts=[start_dim], ends=[end_dim + 1])
+    slice2 = g.op("ReduceProd", slice2)
+    slices = [slice1, slice2]
+    if end_dim < dim - 1:
```

Comment:
yes end_dim can be -1, that's why only if end_dim < dim-1 slice 3 is created

Reply:
but when `end_dim` is -1 `end_dim < dim-1` is true. I guess `slice3` here will still do the right thing, since it will have `starts=0` and `ends=dim`, but if it is not needed then I think we should add `end_dim != -1` to the condition in the "if".

Reply:
that's why there is if end_dim < 0 :end_dim = dim + end_dim in flatten in symbolic_opset 9 and 11:
for example: https://github.com/pytorch/pytorch/pull/40418/files#diff-e5e8501daccb885bc428bcf6c54fd98eL664